content/11-serverless/03-serverless-challenges.mdx

```diff
@@ -0,0 +1,105 @@
+---
+title: 'Introduction to common serverless challenges'
+metaTitle: 'Serverless computing challenges'
+metaDescription: 'Learn about common serverless challenges like cold starts, lock-in, debugging trouble and more.'
+metaImage: '/content/serverless/what-is-serverless/header.png'
+---
+
+## Introduction
+
+The serverless paradigm represents a notable shift in the way that application and web developers interact with infrastructure, language runtimes, and supplemental services.  It offers the freedom to focus on your primary area of concern by abstracting away and taking responsibility for many of the environmental factors that traditionally affect the way code runs in production.
+
+While [serverless computing has many benefits](https://www.prisma.io/dataguide/serverless/what-is-serverless#when-is-serverless-a-good-choice), it also has some challenges that must be acknowledged or addressed before you can be successful.  In this guide, we'll talk about some of the main pain points of the current generation of solutions and discuss what they mean and how you might work around them.  You should come away with a better understanding of what requirements you may have to fulfill and what roadblocks you may run into.
+
+<PrismaOutlinks>
+
+[Prisma's Data Proxy](https://www.prisma.io/docs/concepts/components/prisma-data-platform#prisma-data-proxy), now in early access, is one way to handle connection issues between your serverless applications and backend databases.  It can help manage ephemeral connections from your serverless functions to avoid exhausting your database connection pool.  Check it out now!
+
+</PrismaOutlinks>
+
+## Cold start problems
+
+One of the most commonly discussed challenges when working with serverless is called the *cold start* problem.  While the goal with serverless is allow functions to immediately be executed on demand, there are some scenarios that may result in predictable delays.
```

Comment:
It might be good to link to the Serverless Glossary for things already included like "cold start" just for initial reference so the reader knows that resource is there.

Reply:
Great idea :+1:

Reply:
Ahh, I think this is one of the issues with the way we're doing glossaries right now.  The headers currently aren't generating anchors, which means we can't link directly to them.  I think I'm going to experiment with adding one to the glossary for this term to see if I can get this working though.

Reply:
Shucks.  Well it looks like it strips any manually assigned the `id` fields that I add upon building, so I can't manually add a link to a term.  I guess we'll have to wait until these glossary definitions are built out as a component to add those links.